src/contexts/global.context.tsx

```diff
@@ -0,0 +1,59 @@
+import { createContext, FC, useContext, useState } from "react";
+
+import { Message } from "src/domain";
+
+type SnackbarState =
+  | {
+      status: "closed";
+    }
+  | {
+      status: "open";
+      message: Message;
+    };
+
+interface GlobalContext {
+  snackbar: SnackbarState;
+  openSnackbar: (message: Message) => void;
+  closeSnackbar: () => void;
+}
+
+const globalContextNotReadyErrorMsg = "The global context is not yet ready";
```

Comment:
I like this!

Reply:
I did it to prevent declaring `openSnackbar` and `closeSnackbar` as optional. `openSnackbar` will be used many times and we don't want to check for `undefined` every single time.